pkg/cmd/workflow.go

```diff
@@ -301,6 +301,16 @@ func doDownloadWorkflow(cmd *cobra.Command, args []string) error {
 	body, err := Client.DownloadWorkflow(name)
 
 	if err != nil {
+		if errors.IsClientResponseNotFound(err) {
+			Dialog.Infof(`No file data found for workflow %v
```

Comment:
I'd say this probably still needs to be `Errorf` -- or maybe something in the middle, `Warnf` (which doesn't exist yet)? Using `Infof` will cause this message to go to stdout and a process using the CLI non-interactively won't be able to tell the difference between a successful download and receiving this message. I think the zero exit code is OK though as long as stdout has no content.

Reply:
But it's not an error. It's perfectly valid not to have a workflow file. From the overall perspective of a workflow, it's not an error to not have a file. From a context perspective of explicitly requesting to _download_ a workflow file, I don't personally believe it's an error in that specific situation either, although that's a matter of perspective. The download did not fail. If we retrieve an invalid workflow file, the download should not fail under those circumstances either, and we can't return or display an error about it either. We could possibly warn in both situations, I suppose, but that's starting to split hairs.

This is strictly informational only to the user. I don't see the point of this.

Reply:
As far as returning an error code, that is explicitly a problem right now.

The Github Actions is using the download functionality to determine whether a workflow exists or not (which is probably not the best processing in the first place). If a user creates a workflow from the UI (which won't have a file), an error returned here from the download processing due to a non-existent workflow file would cause that flow to attempt to add a new workflow. Considering that processing is trying to create the workflow as well as the initial file content, that would fail as the workflow already exists. Instead of replacing the file as it should, it would just error out.

Reply:
The GitHub Actions situation makes sense, and I think in that case indeed the zero exit code makes sense.

Unfortunately with POSIX we're stuck with two streams, stdout and stderr, and if users are expecting to get workflow content on stdout putting informational messages on it as well isn't ideal. This doesn't apply to all commands, but I think there's a reasonable expectation that stdout comes through unblemished in this particular case. For example, with this change, `relay workflow download foo | gzip >my-workflow.gz` would end up compressing this message instead of the actual content of the workflow, which should be an empty string. That's why I think we should put the message on stderr instead, especially if we're going to use the zero exit code.

Reply:
Ahh, yes, I see. In this one case, we're already inconsistent. That makes sense.

Reply:
Adding a warning seems fair then.